framework/src/media/utils/MediaUtils.cpp

```diff
@@ -68,10 +72,227 @@ audio_type_t getAudioTypeFromPath(std::string datapath)
 	} else if ((extension.compare("") == 0) || (extension.compare("pcm") == 0) || (extension.compare("raw") == 0)) {
 		medvdbg("audio type : pcm\n");
 		return AUDIO_TYPE_PCM;
+	} else if (extension.compare("wav")) {
+		medvdbg("audio type : wav\n");
+ 		return AUDIO_TYPE_WAVE;
 	} else {
 		medvdbg("audio type : unknown\n");
 		return AUDIO_TYPE_INVALID;
 	}
 }
+
+void mp3_header_parsing(unsigned char * header, unsigned int *channel, unsigned int *sample_rate, audio_format_type_t *pcmFormat)
+{
+	*pcmFormat = AUDIO_FORMAT_TYPE_S16_LE;
+	
+	unsigned char bit;
+	unsigned int MPEG_Version;	
+	bit = header[1];
+	bit >>= 3;
+	bit &= (unsigned char)3;
+
+	if (bit == 0) {
+		MPEG_Version = 2;
+	}
+	else if (bit == 2) {
+		MPEG_Version = 1;
+	}
+	else if (bit == 3) {
+		MPEG_Version = 0;
+	}
+	bit = header[2];
+	bit >>= 2;
+	bit &= 3;
+	
+	if (bit == 0) {
+		*sample_rate = 44100;
+	}
+	else if (bit == 1) {
+		*sample_rate = 48000;
+	}
+	else if (bit == 2) {
+		*sample_rate = 32000;
+	}
+	else{
+		//Reserved
+	}
+	*sample_rate >>= MPEG_Version;
+	bit = header[3];
+	bit >>= 6;
+	if (bit <= 2) {
+		*channel = 2;
+	}
+	else{
+		*channel = 1;
+	}
+
+	return;
+}
+
+void aac_header_parsing(unsigned char * header, unsigned int *channel, unsigned int *sample_rate, audio_format_type_t *pcmFormat)
+{
+	unsigned char bit;
+	
+	*pcmFormat = AUDIO_FORMAT_TYPE_S16_LE;
+
+	bit = header[2];
+	bit >>= 2;
+	bit &= 15;
+
+	switch (bit) {
+	case 0 :
+		*sample_rate = 96000;
+		break;
+	case 1 :
+		*sample_rate = 88200;
+		break;
+	case 2 :
+		*sample_rate = 64000;
+		break;
+	case 3 :
+		*sample_rate = 48000;
+		break;
+	case 4 :
+		*sample_rate = 44100;
+		break;
+	case 5 :
+		*sample_rate = 32000;
+		break;
+	case 6 :
+		*sample_rate = 24000;
+		break;
+	case 7 :
+		*sample_rate = 22050;
+		break;
+	case 8 :
+		*sample_rate = 16000;
+		break;
+	case 9 :
+		*sample_rate = 12000;
+		break;
+	case 10 :
+		*sample_rate = 11025;
+		break;
+	case 11 :
+		*sample_rate = 8000;
+		break;
+	case 12 :
+		*sample_rate = 7350;
+		break;
+	default :
+		//Reserved
```

Comment:
If it comes here, default, it works wrongly I think.
Please Initialize default value first or add error handling code here, If I am right.